File: backend/src/main/java/com/agriconnect/service/AuthService.java
package com.agriconnect.service;

import com.agriconnect.dto.request.ForgotPasswordRequest;
import com.agriconnect.dto.request.LoginRequest;
import com.agriconnect.dto.request.RefreshTokenRequest;
import com.agriconnect.dto.request.RegisterRequest;
import com.agriconnect.dto.request.ResetPasswordRequest;
import com.agriconnect.dto.response.AuthResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);
}

File: frontend/src/services/authService.js
import api from './api';

export const authService = {
  register: (data) => api.post('/auth/register', data),
  login: (data) => api.post('/auth/login', data),
  refreshToken: (data) => api.post('/auth/refresh', data),
  forgotPassword: (data) => api.post('/auth/forgot-password', data),
  resetPassword: (data) => api.post('/auth/reset-password', data),
};
